### ISSUE TITLE
train_samples returned after fine-tuning seems different from the size of the train dataset

### ISSUE THREAD

**Issue by showgood880702** [OptimalScale/LMFlow#337]
deepspeed ${deepspeed_args} \
  examples/finetune.py \
    --model_name_or_path ./lama-7b-hf \
    --dataset_path /data/alpaca/train \
    --output_dir ./finetuned_models --overwrite_output_dir \
    --num_train_epochs 0.01 \
    --learning_rate 1e-5 \
    --block_size 512 \
    --per_device_train_batch_size 1 \
    --use_lora 1 \
    --lora_r 8 \
    --save_aggregated_lora 1\
    --deepspeed configs/ds_config_zero2.json \
    --bf16 \
    --run_name finetune_with_lora \
    --validation_split_percentage 10 \
    --logging_steps 20 \
    --do_train \
    --ddp_timeout 72000 \
    --dataloader_num_workers 1

***** train metrics *****
  epoch                    =       0.01
  train_loss               =     2.0462
  train_runtime            = 0:00:37.62
  train_samples            =       9886
  train_samples_per_second =      2.627
  train_steps_per_second   =      2.631

I tried to fine-tune with the downloaded alpaca train dataset, which has 52k data. But the final train_samples return me 9886. Is this correct?

**Comment by shizhediao** [comment]
Please change `epoch` from 0.01 to 3.
Thanks!

**Comment by showgood880702** [comment]
Thank you for the replying.
Yes I tried with epoch 3, and even tried with different hyperparameters like learning rate and validation split percentage. But the train_samples still returned 9886.
And also I tried fine-tuning with the PubmedQA 210k data. The train_samples returned around 180k.

**Comment by shizhediao** [comment]
Hi,
Thanks for your information. Now I understand your question.
This is as expected because there is a 'group_text' operation in the code that concatenates multiple samples together.  Therefore, the number of samples is smaller than the number of data

**Comment by showgood880702** [comment]
Thank you for the explanation!
